### PR TITLE
feat: use rust hashtree bindings

### PR DESCRIPTION
**Pull request by matthewkeil**
Implement rust bindings in hashtree and update the submodule to use the mac_fix branch to build on mac

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,42 +1,47 @@
 #![deny(clippy::all)]
 
-use std::{ffi::c_ulonglong, os::raw::c_uchar};
+use hashtree_rs::{hash as HASH, init};
```
Comment (wemeetagain) [inline src/lib.rs:3]: It looks like this is published https://crates.io/crates/hashtree-rs

We can simplify by removing the git submodule, custom build.rs step.

File: Cargo.toml [hunk 2]
```diff
@@ -1,19 +1,29 @@
 [package]
-edition = "2021"
 name = "chainsafe_hashtree"
 version = "0.0.0"
+edition = "2021"
+authors = [
+    # TODO: @wemeetagain add your email address here
+    "Cayman Navan",
```
Comment (wemeetagain) [inline Cargo.toml:7]: do we need authors here? this crate won't ever get published, right?

File: src/lib.rs (hunk 1 above)
Comment (matthewkeil) [reply src/lib.rs:3]: The code to fix the Mac and Linux arm builds is not on main yet. Once they are merged we can swap though

File: Cargo.toml (hunk 2 above)
Comment (matthewkeil) [reply Cargo.toml:7]: Hopefully it will 😊. Should speed up hashing in lodestar a lot

Comment (matthewkeil) [reply Cargo.toml:7]: Oh snap, good call. That should go in the package.json. You are right, the rust crate doesn't get published....

**Comment by matthewkeil** [comment]
OK @wemeetagain I've taken this as far as I can for the moment.  I worked on the base issue with windows and got it building for `musl` as well.  There is a problem with the linking of windows for the rust bindings in [prysmaticlabs/hashtree](https://github.com/prysmaticlabs/hashtree) but the C lib builds fine.  Reached out to @potuz about working on a PR in there for that.

File: package.json [hunk 3]
```diff
@@ -17,7 +17,11 @@
     "name": "hashtree",
     "triples": {
       "additional": [
-        "aarch64-unknown-linux-gnu"
+        "x86_64-apple-darwin",
+        "aarch64-apple-darwin",
+        "x86_64-unknown-linux-gnu",
```
Comment (wemeetagain) [inline package.json:27]: can you also include "aarch64-unknown-linux-gnu" (arm linux support)

Comment (matthewkeil) [reply package.json:27]: OK.  linux arm64 works now.  Just need to get windows working properly and we are good to go

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**